src/hw_isolation_record/manager.cpp

```diff
@@ -482,5 +482,53 @@ sdbusplus::message::object_path Manager::createWithEntityPath(
     return *entryPath;
 }
 
+std::optional<std::tuple<entry::EntrySeverity, entry::EntryErrLogPath>>
+    Manager::getIsolatedHwRecordInfo(
+        const sdbusplus::message::object_path& hwInventoryPath)
+{
+    // Make sure whether the given hardware inventory is exists
+    // in the record list.
+    auto entryIt =
+        std::find_if(_isolatedHardwares.begin(), _isolatedHardwares.end(),
+                     [hwInventoryPath](const auto& ele) {
+                         bool isIsolated = false;
+                         for (const auto& assocEle : ele.second->associations())
+                         {
+                             if ((std::get<0>(assocEle) == "isolated_hw") &&
+                                 (std::get<2>(assocEle) == hwInventoryPath.str))
+                             {
+                                 isIsolated = true;
+                                 break;
```

Comment:
return !ele.second->resolved();

can we do this?

Reply:
Ack
```
for (..) 
{
    if (..)
    {
       return !ele.second->resolved();
    }
}

return false;
```